lldb/source/Core/FormatEntity.cpp

```diff
@@ -2204,7 +2204,9 @@ static Status ParseInternal(llvm::StringRef &format, Entry &parent_entry,
                   return error;
                 }
               } else if (FormatManager::GetFormatFromCString(
-                             entry.printf_format.c_str(), true, entry.fmt)) {
+                             entry.printf_format.c_str(), true,
+                             entry.fmt)) { // Try GetFormatFromCString again,
```

Comment:
Is the ordering important here? Why not try partial formatting first from the start? The code paths are nearly identical except for one parameter... The comment definitely helps disambiguate but it would be nicer if we could just rewrite this entirely.

Reply:
doing partial from the outset would incorrectly match `%V` as `void` – the correct behavior is to print the lldb **V**alue. It's definitely not an ideal setup.

Reply:
In summary, the order is: exact printf format > lldb format > partial printf format

Reply:
I guess if we assume that there's no overlap between printf and lldb format (which there shouldn't be) then the lldb format can be done first, and the printf can be done second, as partial.

Reply:
Pushed changes and updated the PR description.